kube/src/client/auth/refreshing_token.rs

```diff
@@ -4,61 +4,57 @@ use std::{
 };
 
 use futures::{ready, Future};
-use http::{header::AUTHORIZATION, Request};
-use hyper::Body;
+use http::{header::AUTHORIZATION, Request, Response};
 use pin_project::pin_project;
 use tower::{layer::Layer, BoxError, Service};
 
 use super::RefreshableToken;
 use crate::Result;
 
-/// `Layer` to decorate the request with `Authorization` header.
-pub struct AuthLayer {
-    auth: RefreshableToken,
+// TODO Come up with a better name
```

Comment:
`RefreshTokenLayer` is probably ok.
`RefreshingTokenLayer` is funny though; like, wow, very original take on a token layer.